File: src/main/java/io/dudek/florystyka/db/QuestionRepository.java
package io.dudek.florystyka.db;

import io.dudek.florystyka.domain.Question;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface QuestionRepository extends JpaRepository<Question, Integer>, TestQuestion {

    Question findQuestionById(Integer id);
    Integer countQuestionByIdIsGreaterThanEqual(Integer id);
    List<Question> findAllById(Iterable<Integer> ids);


}

File: src/main/kotlin/pl/quizwiz/question/domain/port/QuestionRepository.kt
package pl.quizwiz.question.domain.port

import pl.quizwiz.question.domain.model.Question
import pl.quizwiz.question.domain.model.QuestionId

interface QuestionRepository {

    fun findQuestionById(id: QuestionId): Question

    fun saveQuestion(question: Question)

}